README.md
# 前端项目脚手架

> 一个基于 vue_cli4 和 nodejs_koa2 的前端项目快速启动脚手架

---

## 一、技术栈

### I. 视图层

- `vue3`
- `typescript`
- `vue-router` - `history mode`
- `elementui-plus`

### II. 数据层

- `koa2`
- `mongoose`
- `基于装饰器封装的控制层`
- `函数式编程` - `Ramda`
- `统一格式化返回`

### III. 前端

- `动态路由`
- `支持基于RBAC权限控制格式的视图控制`
- `Commit 规范` - `基于业内最流行的标准`
- `本地开发环境配置`
- `明确项目文件夹结构`



## 二、项目使用
### I. 依赖安装

- `npm install`

### II. 本地开发

- `npm run dev-s`
  - `保证 node 版本在 10.14 以上`
- `npm run serve`
  - `保证 node 版本在 12.15 以上`
- 访问 `localhost:8080`


### III.生产环境

- `npm run build`
    - `保证 node 版本在 10.14 以上`
- `npm run release`
- 访问 `localhost:3000/stage`


### IV. 项目提交

- 【前置】全局安装 `commitizen`
- 【前置】选择适配器 `echo '{ "path": "cz-conventional-changelog" }' > ~/.czrc`
- `npm run commit`或`git cz`
- 请保证严格依赖 `git commit 标准`


CHANGELOG.md
## 0.2.9 (2021-03-15)


### Features

* add breadCrumb view ([7f4a686](https://github.com/hellobeifeng/vue3-koa2-stage/commit/7f4a686e6dc10b23ada464b93f80c37f87994ded))
* add element-plus & set vue-router lazy load ([0099023](https://github.com/hellobeifeng/vue3-koa2-stage/commit/009902354efb88421f4b98df70b9075f345c288e))
* add mongose  && bugfix ([3f5038b](https://github.com/hellobeifeng/vue3-koa2-stage/commit/3f5038b4572d192b05debc84cf475c6375be3b7a))
* add server cookie-session ([28d9896](https://github.com/hellobeifeng/vue3-koa2-stage/commit/28d98965291036a966323ae6219afdfde9dac46b))
* add server/log ([e0dcf2e](https://github.com/hellobeifeng/vue3-koa2-stage/commit/e0dcf2e39e768ea532494158a6333904787962e9))
* add vuex&vue-router ([5b8f990](https://github.com/hellobeifeng/vue3-koa2-stage/commit/5b8f990215e8b9b4c166132546b102f59b4744e8))
* client dir & vue.config.js ([500e4f4](https://github.com/hellobeifeng/vue3-koa2-stage/commit/500e4f480d0ce8538a4dddc8d8672676130dd80a))
* client/api  proxy server/api successfully git add . ([974b07f](https://github.com/hellobeifeng/vue3-koa2-stage/commit/974b07f99d84fff66b48a7060d032c58b5ee549e))
* client/dist mount on server /stage/ ([b9a7519](https://github.com/hellobeifeng/vue3-koa2-stage/commit/b9a7519caba17b9c2728237b5f9d80d494bb844b))
* components: menubar && menubar-item ([a4ecd85](https://github.com/hellobeifeng/vue3-koa2-stage/commit/a4ecd85b3d84709918383daa9c8d5d59c63b25e9))
* decorator restructure server/router git add . && add editorconfig && bugfix ([4573d60](https://github.com/hellobeifeng/vue3-koa2-stage/commit/4573d60dc15c4e77dfe11e49e638f074b91b1ce1))
* init server/ ([96d8e84](https://github.com/hellobeifeng/vue3-koa2-stage/commit/96d8e84b242c65129985f7a014a82daa56a4b79d))
* project init ([961dfc0](https://github.com/hellobeifeng/vue3-koa2-stage/commit/961dfc030c3e7d6d24efc460847c9a5b79394090))
* **commitlint:** add commt lint ([1320952](https://github.com/hellobeifeng/vue3-koa2-stage/commit/13209521f93d03cd34490c72f3351d04371acc2e))
* **package.json:** 增加提交日志脚本 ([f0d5afe](https://github.com/hellobeifeng/vue3-koa2-stage/commit/f0d5afe15d1af1571907b6cf2174d3c2b16572a9))
* server support decorators env ([44526a9](https://github.com/hellobeifeng/vue3-koa2-stage/commit/44526a9896caf29aa9ba0fc4e54e02d745a48f04))
* server/ add controller/ service and response formatter ([781b672](https://github.com/hellobeifeng/vue3-koa2-stage/commit/781b67291ea82c1bfc6374ebfa2cfa980c1df51d))
* vue-router support history mode git add . ([fd55f69](https://github.com/hellobeifeng/vue3-koa2-stage/commit/fd55f69187e8db37b6da8ee405a56183497e6576))
* vuex/module && loading/message componennts ([f5cdcd9](https://github.com/hellobeifeng/vue3-koa2-stage/commit/f5cdcd9cde656893a51f002fc26e365dba165c8e))
* 基于rbac mock数据的菜单权限 ([8ce4ea9](https://github.com/hellobeifeng/vue3-koa2-stage/commit/8ce4ea98c6ae5425266b7ee0f4f1ef43ac9442de))
* 支持动态路由写法 ([dd62860](https://github.com/hellobeifeng/vue3-koa2-stage/commit/dd628607a9b0687c5157224cc9d26090f72538e1))



# 0.1.0 (2021-03-15)


### Features

* add breadCrumb view ([7f4a686](https://github.com/hellobeifeng/vue3-koa2-stage/commit/7f4a686e6dc10b23ada464b93f80c37f87994ded))
* add element-plus & set vue-router lazy load ([0099023](https://github.com/hellobeifeng/vue3-koa2-stage/commit/009902354efb88421f4b98df70b9075f345c288e))
* add mongose  && bugfix ([3f5038b](https://github.com/hellobeifeng/vue3-koa2-stage/commit/3f5038b4572d192b05debc84cf475c6375be3b7a))
* add server cookie-session ([28d9896](https://github.com/hellobeifeng/vue3-koa2-stage/commit/28d98965291036a966323ae6219afdfde9dac46b))
* add server/log ([e0dcf2e](https://github.com/hellobeifeng/vue3-koa2-stage/commit/e0dcf2e39e768ea532494158a6333904787962e9))
* add vuex&vue-router ([5b8f990](https://github.com/hellobeifeng/vue3-koa2-stage/commit/5b8f990215e8b9b4c166132546b102f59b4744e8))
* client dir & vue.config.js ([500e4f4](https://github.com/hellobeifeng/vue3-koa2-stage/commit/500e4f480d0ce8538a4dddc8d8672676130dd80a))
* client/api  proxy server/api successfully git add . ([974b07f](https://github.com/hellobeifeng/vue3-koa2-stage/commit/974b07f99d84fff66b48a7060d032c58b5ee549e))
* client/dist mount on server /stage/ ([b9a7519](https://github.com/hellobeifeng/vue3-koa2-stage/commit/b9a7519caba17b9c2728237b5f9d80d494bb844b))
* components: menubar && menubar-item ([a4ecd85](https://github.com/hellobeifeng/vue3-koa2-stage/commit/a4ecd85b3d84709918383daa9c8d5d59c63b25e9))
* decorator restructure server/router git add . && add editorconfig && bugfix ([4573d60](https://github.com/hellobeifeng/vue3-koa2-stage/commit/4573d60dc15c4e77dfe11e49e638f074b91b1ce1))
* init server/ ([96d8e84](https://github.com/hellobeifeng/vue3-koa2-stage/commit/96d8e84b242c65129985f7a014a82daa56a4b79d))
* project init ([961dfc0](https://github.com/hellobeifeng/vue3-koa2-stage/commit/961dfc030c3e7d6d24efc460847c9a5b79394090))
* **commitlint:** add commt lint ([1320952](https://github.com/hellobeifeng/vue3-koa2-stage/commit/13209521f93d03cd34490c72f3351d04371acc2e))
* server support decorators env ([44526a9](https://github.com/hellobeifeng/vue3-koa2-stage/commit/44526a9896caf29aa9ba0fc4e54e02d745a48f04))
* server/ add controller/ service and response formatter ([781b672](https://github.com/hellobeifeng/vue3-koa2-stage/commit/781b67291ea82c1bfc6374ebfa2cfa980c1df51d))
* vue-router support history mode git add . ([fd55f69](https://github.com/hellobeifeng/vue3-koa2-stage/commit/fd55f69187e8db37b6da8ee405a56183497e6576))
* vuex/module && loading/message componennts ([f5cdcd9](https://github.com/hellobeifeng/vue3-koa2-stage/commit/f5cdcd9cde656893a51f002fc26e365dba165c8e))
* 基于rbac mock数据的菜单权限 ([8ce4ea9](https://github.com/hellobeifeng/vue3-koa2-stage/commit/8ce4ea98c6ae5425266b7ee0f4f1ef43ac9442de))
* 支持动态路由写法 ([dd62860](https://github.com/hellobeifeng/vue3-koa2-stage/commit/dd628607a9b0687c5157224cc9d26090f72538e1))





vue.config.js
const path = require("path");
const CopyPlugin = require("copy-webpack-plugin")
function resolve(dir) {
  return path.join(__dirname, dir);
}

module.exports = {
  devServer:{
    host:'localhost',
    port:8080,
    proxy:{
      '/stage':{
        target:'http://localhost:3000',
        changeOrigin:true
      }
    }
  },
  pages: {
    index: {
      entry: './client/src/main.ts',
      template: resolve("./client/public/index.html"),
      title: 'Vue3_Koa2 脚手架'
    }
  },
  outputDir: resolve('client/dist'),
  publicPath: process.env.NODE_ENV === 'production' ? '/stage/' : '/',
  lintOnSave: true,
  productionSourceMap: false,
  chainWebpack: config => {
    config.resolve.alias
      .set("@components", resolve("./client/src/components"))
      .set("@views", resolve("./client/src/views"))
      .set("@hooks", resolve("./client/src/hooks"))
  },
  configureWebpack: {
    plugins: [
      new CopyPlugin(
        [
          {
            from: resolve("./client/public"),
            to: resolve("./client/dist"),
            toType: 'dir',
            ignore: [
              '.DS_Store',
              {
                glob: 'index.html',
                matchBase: false
              }
            ]
          }
        ]
      )
    ]
  }
}
